src/xmlsec/key_manager.rs

```diff
@@ -0,0 +1,110 @@
+use super::XmlSecError;
+use super::XmlSecKey;
+use super::XmlSecKeyDataStore;
+use super::XmlSecKeyInfoContext;
+use super::XmlSecKeyStore;
+use super::XmlSecResult;
+use crate::bindings::xmlSecKeyDataStoreId;
+use crate::bindings::{
+    xmlSecKeysMngrCreate, xmlSecKeysMngrDestroy, xmlSecKeysMngrPtr,
+    xmlSecOpenSSLAppDefaultKeysMngrAdoptKey, xmlSecOpenSSLAppDefaultKeysMngrInit,
+};
+
+/// The key manager type that's used for handling keys that are used to do
+/// things like encrypt other keys.
+pub struct XmlSecKeyManager {
+    mngr: xmlSecKeysMngrPtr,
+}
+
+impl XmlSecKeyManager {
+    /// Create and initialize a new key manager.
+    pub fn new() -> XmlSecResult<Self> {
+        unsafe {
+            let mngr = xmlSecKeysMngrCreate();
+            if mngr.is_null() {
+                return Err(XmlSecError::KeyManagerCreateFailure);
+            }
+            if xmlSecOpenSSLAppDefaultKeysMngrInit(mngr) != 0 {
+                return Err(XmlSecError::KeyManagerDefaultInitFailure);
+            }
+            Ok(Self { mngr })
+        }
+    }
+
+    /// Returns the internal pointer to the type.
+    pub fn as_ptr(&self) -> xmlSecKeysMngrPtr {
+        self.mngr
+    }
+
+    // xmlSecKeyPtr	xmlSecKeysMngrFindKey ()
+    /// Lookups key in the keys manager keys store. The caller is responsible
+    /// for destroying the returned key using xmlSecKeyDestroy method.
+    pub fn find_key(&self, _name: &str, _ctx: &XmlSecKeyInfoContext) -> XmlSecResult<XmlSecKey> {
+        todo!()
+    }
+
+    /// add key to keys manager, from now on keys manager is responsible for
+    /// destroying key.
+    pub fn adopt_key(&self, key: XmlSecKey) -> XmlSecResult<()> {
+        unsafe {
+            let key = XmlSecKey::leak(key);
+            if xmlSecOpenSSLAppDefaultKeysMngrAdoptKey(self.mngr, key) != 0 {
+                return Err(XmlSecError::KeyManagerKeyAdoptionFailure);
+            }
+        }
+        Ok(())
+    }
+
+    // xmlSecKeyPtr	xmlSecKeysMngrFindKeyFromX509Data ()
+    // /// Lookups key in the keys manager keys store. The caller is responsible
+    // /// for destroying the returned key using xmlSecKeyDestroy method.
+    // fn find_key_from_x509(&self) -> XmlSecResult<XmlSecKey> {
+    //     // TODO: Eventually add an implementation for an x509 certificate
+    //     // that we can leverage from inside of xml sec.
+    //     todo!()
+    // }
+
+    /// Adopts keys store in the keys manager.
```

Comment:
I see these functions are implemented as `todo!()`. Are we planning to implement them later? If so, I'd prefer if we deleted these implementations to make sure someone doesn't call them by accident.

Reply:
Removed.